### PR TITLE
Add tuning documentation

### PR DESCRIPTION
**Pull request by wilfriedroset**
#### What this PR does

This commit is a followup of the PR #9149 where we discussed the idea of centralizing the known tuning associated with their use cases.

#### Which issue(s) this PR fixes or relates to

Relates to #9149

#### Checklist

- [ ] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by wilfriedroset** [comment]
ping @56quarters @bboreham for the review as you were helping on the proposal 🙏🏾

File: docs/sources/mimir/configure/tuning.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+- Make sure `wal_compression_enabled` is not enable
+- Make sure `grpc_compression` is either off which is the default or configured to `snappy`. `gzip` consumes more CPU than `snappy`. However, disabling `grpc_compression` implies more network traffic and in turn might increase the total cost of ownership (TCO) of running Grafana Mimir.
+
+If users must used compression, like for example to fit in the network bandwidth, they might consider using nodes with more powerful CPU. This implies a increase of the TCO.
+
+## Increase the cache size on a budget
+
+Grafana Mimir relies on Memcached for its caches. Memcached relies, by default only on the memory.
+Similarly to the work of the Ops team behind Grafana Loki, one could enable the `extstore` feature of Memcached.
```
Comment (tacole02) [inline docs/sources/mimir/configure/tuning.md:37]: I don't think it's necessary to include what the Loki team is doing here. Rather, could we provide a brief description as to what enabling `extstore` accomplishes?

File: docs/sources/mimir/configure/tuning.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+See: [how we scaled Grafana Cloud Logs' Memcached cluster to 50TB and improved reliability](https://grafana.com/blog/2023/08/23/how-we-scaled-grafana-cloud-logs-memcached-cluster-to-50tb-and-improved-reliability/)
```
Comment (tacole02) [inline docs/sources/mimir/configure/tuning.md:39]:
```suggestion
Refer to [how we scaled Grafana Cloud Logs' Memcached cluster to 50TB and improved reliability](https://grafana.com/blog/2023/08/23/how-we-scaled-grafana-cloud-logs-memcached-cluster-to-50tb-and-improved-reliability/).
```

File: docs/sources/mimir/configure/tuning.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+## Periodic latency spikes when cutting blocks
+
+Depending on the workload, users might witness latency spikes when Grafana Mimir cuts blocks.
```
Comment (tacole02) [inline docs/sources/mimir/configure/tuning.md:43]:
```suggestion
Depending on the workload, you might witness latency spikes when Mimir cuts blocks.
```

File: docs/sources/mimir/configure/tuning.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+To reduce the impacts of this behavior, users might consider:
```
Comment (tacole02) [inline docs/sources/mimir/configure/tuning.md:44]:
```suggestion
To reduce the impact of this behavior, consider the following:
```

File: docs/sources/mimir/configure/tuning.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+- Upgrade to `2.15+`, See: <https://github.com/grafana/mimir/commit/03f2f06e1247e997a0246d72f5c2c1fd9bd386df>
```
Comment (tacole02) [inline docs/sources/mimir/configure/tuning.md:46]:
```suggestion
- Upgrade to `2.15+`. Refer to <https://github.com/grafana/mimir/commit/03f2f06e1247e997a0246d72f5c2c1fd9bd386df>.
```

File: docs/sources/mimir/configure/tuning.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+- Reduce `-blocks-storage.tsdb.block-ranges-period`, default `2h`, For example try `1h`
```
Comment (tacole02) [inline docs/sources/mimir/configure/tuning.md:47]:
```suggestion
- Reduce `-blocks-storage.tsdb.block-ranges-period`, default `2h`. For example. try `1h`.
```

**Comment by wilfriedroset** [comment]
thank you @tacole02 for your review I've taken into account your remarks.

**Review by dimitarvdimitrov — COMMENTED** [review]
there's already the "Production tips" [article](https://github.com/grafana/mimir/blob/a817c626a2c31359476dfcffa9700cab1d8af74c/docs/sources/mimir/manage/run-production-environment/production-tips/index.md) does it make sense to merge this content there?

**Review by dimitarvdimitrov — APPROVED** [review]
Thank you for writing this ❤️

**Pull request merged**